From 9657e707c85ad6d1159320099f1af1dd3d2cdc2b Mon Sep 17 00:00:00 2001
From: Bogdan-Gabriel Roman <bogdan-gabriel.roman@nxp.com>
Date: Wed, 5 Jan 2022 13:45:57 +0200
Subject: [PATCH 6/6] wkpu: remove GPIO166 external wake-up source

The GPIO166 pin is used as external wake-up source on RDB1 platforms
which are not supported anymore. On newer RDB platforms it is not
dedicated to the wake-up functionality. Remove this wake-up source as it
leads to inconsistent behaviour during suspend-and-resume.

Issue: ALB-8238
Upstream-Status: Pending 

Signed-off-by: Bogdan-Gabriel Roman <bogdan-gabriel.roman@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 fdts/fsl-s32-gen1.dtsi | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/fdts/fsl-s32-gen1.dtsi b/fdts/fsl-s32-gen1.dtsi
index c3aa51ee3..2b5cfd220 100644
--- a/fdts/fsl-s32-gen1.dtsi
+++ b/fdts/fsl-s32-gen1.dtsi
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020-2021 NXP
+ * Copyright 2020-2022 NXP
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -137,14 +137,10 @@
 		reg = <0x0 0x40090000 0x0 0x10000>, /* WKPU */
 		      <0x0 0x4007cb04 0x0 0x4>; /* S32G_STDBY_GPR */
 		/*
-		 * Enable RTC and GPIO166 (J5, pin 30 on S32G274A RDB REV B)
-		 * as wake-up sources
+		 * Enable RTC as wake-up source
 		 */
 		nxp,irqs = <S32GEN1_WKPU_RTC_IRQ
 			    S32GEN1_WKPU_IRQ_RISING
-			    S32GEN1_WKPU_PULL_DIS>,
-			   <S32GEN1_WKPU_EXT_IRQ(11)
-			    S32GEN1_WKPU_IRQ_FALLING
 			    S32GEN1_WKPU_PULL_DIS>;
 
 		nxp,warm-boot = <S32GEN1_WKPU_LONG_BOOT>;
-- 
2.17.1

